### ISSUE TITLE
LLVM crashes when building for Solana

### ISSUE THREAD

**Issue by LucasSte** [hyperledger-labs/solang#789]
The following contract crashes LLVM when targeting Solana. It works, however, if we target Substrate.

```solidity
contract foo {
    struct S { int f1; }
        S[] arr;

    function g() private returns (S storage, S storage) {
        return (arr[0], arr[1]);
    }

    function f() public returns (S, S) {
        S[] storage ptrArr = arr;
        ptrArr.push(S({f1: 1}));
        ptrArr.push(S({f1: 2}));
        return g();
    }
}
```

Error:
```
Assertion failed: (cast<PointerType>(getOperand(1)->getType()) ->isOpaqueOrPointeeTypeMatches(getOperand(0)->getType()) && "Ptr must be a pointer to Val type!"), function AssertOK, file Instructions.cpp, line 1482.
```

**Comment by LucasSte** [comment]
The simplest failing example is the following:

```solidity
contract foo {
    struct S { int f1; }
        S[] arr;

    function g() private view returns (S storage) {
        return arr[0];
    }
}
```

**Comment by LucasSte** [comment]
Fixed by #825 .
